### ISSUE TITLE
Fresh-Install -  PREFIX/share/next - "No such file or directory"

### ISSUE THREAD

**Issue by lxbarbosa** [atlas-engineer/next#626]
**Describe the bug**
For some reason, Next(build from source) is not creating the folder where next.socket lives in. I had to create it so to Next start. 

Aint MAKEFILE lacking a `mkdir -p "$(DESTDIR)$(DATADIR)/next` in `install-assets`? 

**Precise recipe to reproduce the issue**
```
 git clone https://github.com/atlas-engineer/next && cd next
make all NEXT_INTERNAL_QUICKLISP=true  PREFIX=~/.local
make install PREFIX=~/.local
```

**Information**
- OS name+version:  Debian SID 11 
- Kernel name+version: 5.4.0-4-amd64
- Graphics card and driver: Intel Graphics - Thinkpad t430
- Desktop environment / Window manager name+version: Qtile 0.14.2
- Next version (from =M-x next-version= or =next --version=): Next 1.5.0-dde1693a
- How you installed Next (Guix pack, package manager, build from source): source
- Lisp implementation/version (if built from source):  SBCL 2.0.2.debian
- WebKitGTK+ | QtWebEngine version:  2.28.0-2

**Output when started from a shell**
```
Next version 1.5.0-dde1693a                                                                                                             
 <INFO> [08:49:23] next start.lisp (listen-socket) -                                                                                    
  Listening on socket "/home/lxbarbosa/.local/share/next/next.socket"
Unhandled IOLIB/SYSCALLS:ENOENT in thread #<SB-THREAD:THREAD "Anonymous thread" RUNNING
                                             {1006A56DC3}>:
  #<Syscall "bind" signalled error ENOENT(2) "No such file or directory" FD=8>

Backtrace for: #<SB-THREAD:THREAD "Anonymous thread" RUNNING {1006A56DC3}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<Syscall "bind" signalled error ENOENT(2) "No such file or directory" FD=8> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* #<Syscall "bind" signalled error ENOENT(2) "No such file or directory" FD=8>)
2: (INVOKE-DEBUGGER #<Syscall "bind" signalled error ENOENT(2) "No such file or directory" FD=8>)
3: (ERROR #<Syscall "bind" signalled error ENOENT(2) "No such file or directory" FD=8>)
4: ((:METHOD IOLIB/SOCKETS:BIND-ADDRESS (IOLIB/SOCKETS:LOCAL-SOCKET IOLIB/SOCKETS:LOCAL-ADDRESS)) #<passive local stream socket, unbound {1007185353}> #<Unix socket address: "/home/lxbarbosa/.local/share/next/next.socket">) [fast-method]
5: ((SB-PCL::EMF IOLIB/SOCKETS:BIND-ADDRESS) #<unused argument> #<unused argument> #<passive local stream socket, unbound {1007185353}> #<Unix socket address: "/home/lxbarbosa/.local/share/next/next.socket"> :REUSE-ADDRESS T)
6: (IOLIB/SOCKETS::%%INIT-SOCKET/LOCAL-STREAM-PASSIVE #<passive local stream socket, unbound {1007185353}> "/home/lxbarbosa/.local/share/next/next.socket" T 5)
7: (NEXT::LISTEN-SOCKET)
8: ((LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS))
9: ((FLET SB-UNIX::BODY :IN SB-THREAD::NEW-LISP-THREAD-TRAMPOLINE))
10: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::NEW-LISP-THREAD-TRAMPOLINE))
11: ((FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::NEW-LISP-THREAD-TRAMPOLINE))
12: ((FLET "WITHOUT-INTERRUPTS-BODY-1" :IN SB-THREAD::CALL-WITH-MUTEX))
13: (SB-THREAD::CALL-WITH-MUTEX #<CLOSURE (FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::NEW-LISP-THREAD-TRAMPOLINE) {7FB39555ED7B}> #<SB-THREAD:MUTEX "thread result lock" owner: #<SB-THREAD:THREAD "Anonymous thread" RUNNING {1006A56DC3}>> NIL T NIL)
14: (SB-THREAD::NEW-LISP-THREAD-TRAMPOLINE #<SB-THREAD:THREAD "Anonymous thread" RUNNING {1006A56DC3}> NIL #<CLOSURE (LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS) {1006A56D6B}> NIL)
15: ("foreign function: call_into_lisp")
16: ("foreign function: new_thread_trampoline")

unhandled condition in --disable-debugger mode, quitting
; 
; compilation unit aborted
;   caught 1 fatal ERROR condition 
```

**Comment by jmercouris** [comment]
Thank you for the report, we are probably not invoking `ensure-directory-exists` before making the socket file!

**Comment by Ambrevar** [comment]
Indeed, I've fixed it, pushing soon.

**Comment by Ambrevar** [comment]
Thanks for the report, @lxbarbosa!
Fixed in 38731b58c07a1a2163f3af02ad0316d7172ef7e6.
Feel free to reopen if it's still an issue.

Also note that the program itself must make sure that the folders exist, not the Makefile, because the folder could very well be removed after compilation.
